### ISSUE TITLE
Cannot search for songs with less than 3-character names (affects Chinese & logographic languages) 

### ISSUE THREAD

**Issue by liangGTY** [citelao/Spotify-for-Alfred#133]

**Comment by liangGTY** [comment]
and I found it impossible to search in Chinese, but english is work

**Comment by citelao** [comment]
你好!

Unfortunately, I do not know how to set up a SOCKS proxy.

I assume you are trying to use this from a country where Spotify does not work. I have never used Spotify in a place like that. I don't think I would be a very good help. You might try a VPN.

I am updating this issue to focus on your second problem, that you cannot search in Chinese.

Now, it looks like Chinese results appear properly:

<img width="623" alt="Chinese results in Spotifious" src="https://user-images.githubusercontent.com/432939/74892881-b7357380-533f-11ea-82c9-d0f32d5a480a.png">

And I can search for mixed Chinese/English content:

<img width="623" alt="Chinese/English search" src="https://user-images.githubusercontent.com/432939/74892951-f2d03d80-533f-11ea-8b6c-cad9dfe35f46.png">

But all-Chinese does not work:

<img width="623" alt="All-Chinese search" src="https://user-images.githubusercontent.com/432939/74892991-109da280-5340-11ea-8355-d1662db4464c.png">

Spotifious is generating output, but it's not appearing. I'm investigating to see what's up.

**Comment by citelao** [comment]
Actually, it looks like some Chinese searches work:

<img width="623" alt="Ni dong bu dong works" src="https://user-images.githubusercontent.com/432939/74893610-a0901c00-5341-11ea-91fe-064c385397e8.png">

I think it's because the first result has unescaped `<` and `>`s in it. Oops! I'll look into that.

**Comment by citelao** [comment]
Yup. Of course I'm not escaping properly. If Spotify is searching properly (which it might not be, since you might need a proxy), then Chinese should properly appear.

However, the search I chose had a bad result in it (a result that had `<` and `>` in it).

You can give it a try by changing the code in your version of Spotifious if you'd like. In `src/citelao/OhAlfred/OhAlfred.php`, around line 262, add some escaping to `escapeQuery`:

```php
     public function escapeQuery($text) {
		$text = str_replace("<", "&lt;", $text);
		$text = str_replace(">", "&rt;", $text);
		$text = str_replace("&", "&amp;", $text);
		$text = str_replace("'", "&#39;", $text);

		return $text;
	}
```

You're specifically adding the lines:

```php
		$text = str_replace("<", "&lt;", $text);
		$text = str_replace(">", "&rt;", $text);
```

(Yes, I know I should use `$text = htmlspecialchars($text);`, but this is a minimal fix for demo).

Good luck!

**Comment by liangGTY** [comment]
Thank you very much for your help, I added these two lines of escape code but still not work, you can try searching for "大鱼".
I also tried debugging, but it failed because I was not familiar with PHP and alfred workflow.

**Comment by citelao** [comment]
Oh, that's because Spotifious requires > 3 characters to search. If you have two more relevant characters, you should be able to search.

**Comment by citelao** [comment]
That was to make searching faster by skipping extra queries (since in English, your query is probably more than 3 characters long). It doesn't make sense for languages with one character per word.

I wonder if PHP can be smarter about it...

**Comment by liangGTY** [comment]
ok.. i will try modify the code to support short char. thank you~

**Comment by citelao** [comment]
I will also look into mitigations so that languages with one-character words are usable. I don't know if I can remove the 3-char limit in general, since I think that might send too many searches in English, but I might be able to detect Chinese at least and remove the limit in that case.

Normally I would use a [break iterator](https://www.php.net/manual/en/class.intlbreakiterator.php) from PHP's `intl` extension, but Mac does not ship with that extension by default.

Reopening to track that work.

**Comment by liangGTY** [comment]
I think you can control whether the search starts or not
